platform/broadcom/sonic-platform-modules-cel/haliburton/scripts/cpu_wdt

```diff
@@ -0,0 +1,303 @@
+DEFAULT_KEEPALIVE = 60
+
+
+# ========================== Syslog wrappers ==========================
+
+
+def log_info(msg, also_print_to_console=False):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_INFO, msg)
+    syslog.closelog()
+
+    if also_print_to_console:
+        print msg
+
+
+def log_warning(msg, also_print_to_console=False):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_WARNING, msg)
+    syslog.closelog()
+
+    if also_print_to_console:
+        print msg
+
+
+def log_error(msg, also_print_to_console=False):
+    syslog.openlog(SYSLOG_IDENTIFIER)
+    syslog.syslog(syslog.LOG_ERR, msg)
+    syslog.closelog()
+
+    if also_print_to_console:
+        print msg
+
+
+class Watchdog(WatchdogBase):
+
+    def __init__(self):
+        # Init cpld reg path
+        self.setreg_path = os.path.join(PLATFORM_CPLD_PATH, SETREG_FILE)
+        self.getreg_path = os.path.join(PLATFORM_CPLD_PATH, GETREG_FILE)
+
+        self.mmc_v = self._get_mmc_version()
+        self.cpld_info = V06_CPLD_WDT_INFO if self.mmc_v <= V06_MMC_VERSION else CPLD_WDT_INFO
+
+        # Set default value
+        self._disable()
+        self.armed = False
+        self.timeout = 0
+
+    def _get_mmc_version(self):
+        hex_str_v = self._get_register_value(MMC_VERSION_REG)
+        return int(hex_str_v, 16)
+
+    def _get_register_value(self, register):
+        # Retrieves the value in the cpld register.
+        self._write_reg(self.getreg_path, register)
+        return self._read_reg(self.getreg_path)
+
+    def _write_reg(self, file_path, value):
+        with open(file_path, 'w') as fd:
+            fd.write(str(value))
+
+    def _read_reg(self, path):
+        with open(path, 'r') as fd:
+            output = fd.readline()
+        return output.strip('\n')
+
+    def _get_level_hex(self, sub_hex):
+        sub_hex_str = sub_hex.replace("x", "0")
+        return hex(int(sub_hex_str, 16))
+
+    def _seconds_to_lmh_hex(self, seconds):
+        ms = seconds*1000  # calculate timeout in ms format
+        hex_str = hex(ms)
+        l = self._get_level_hex(hex_str[-2:])
+        m = self._get_level_hex(hex_str[-4:-2])
+        h = self._get_level_hex(hex_str[-6:-4])
+        return (l, m, h)
+
+    def _enable(self):
+        """
+        Turn on the watchdog timer
+        """
+        enable_val = '{} {}'.format(
+            self.cpld_info['wdt_en_reg'], self.cpld_info['wdt_en_cmd'])
+        return self._write_reg(self.setreg_path, enable_val)
+
+    def _disable(self):
+        """
+        Turn off the watchdog timer
+        """
+        disable_val = '{} {}'.format(
+            self.cpld_info['wdt_en_reg'], self.cpld_info['wdt_dis_cmd'])
+        return self._write_reg(self.setreg_path, disable_val)
+
+    def _keepalive(self):
+        """
+        Keep alive watchdog timer
+        """
+        if self.mmc_v <= V06_MMC_VERSION:
+            self._disable()
+            time.sleep(0.1)
```

Comment:
why sleep here?

Reply:
removed in https://github.com/Azure/sonic-buildimage/pull/6259/commits/c5344ce50a9a1371e6a06b34d7c880f3fbc34510